lib/runtime-c-api/crates/wasm-c-api/src/lib.rs

```diff
@@ -0,0 +1,1355 @@
+//! entrypoints for the standard C API
+//! TODO: come up with a better name than `c_api`.
+
+use std::convert::{TryFrom, TryInto};
+use std::ffi::c_void;
+use std::mem;
+use std::ptr;
+use std::slice;
+use std::sync::Arc;
+
+use wasmer::compiler::compile;
+use wasmer::error::{CallError, RuntimeError};
+use wasmer::import::{ImportObject, LikeNamespace, Namespace};
+use wasmer::module::Module;
+use wasmer::units;
+use wasmer::wasm;
+
+// TODO: remove delete from macro generation, need to do that manually
+
+// TODO: investigate marking things like the C++ API does (does not work in return position)
```

Comment:
What do you mean?

Reply:
In the C++ API they use `own<T>` to represent the transfer of ownership ; I wasn't able to get this to work in the return position using `#[repr(transparent)]` in Rust